### ISSUE TITLE
register_model doesn't seem to work

### ISSUE THREAD

**Issue by TheDJVG** [justengel/pydantic_decoder#1]
I found this project and it's exactly what I need when exchanging pydantic models between systems. Unfortunately I'm not able to get `register_model` to work and the example doesn't seem to work either:

```python
from pydantic import BaseModel
from pydantic_decoder import register_model, encode, decode

class MyModel(BaseModel):
    first_name: str
    last_name: str

class MyModel2(BaseModel):
    first_name: str
    last_name: str


register_model(MyModel)
register_model('Model2', MyModel2)

model = MyModel(first_name='John', last_name='Doe')
msg = encode(model)
m = decode(msg)
assert m == model
assert isinstance(m, MyModel)

model2 = MyModel2(first_name='John', last_name='Doe')
msg = encode(model2)
m2 = decode(msg)
assert m2 == model2
assert isinstance(m2, MyModel2)
```
```bash
Traceback (most recent call last):
  File "/home/djvg/.config/JetBrains/PyCharm2020.3/scratches/scratch_6.py", line 18, in <module>
    m = decode(msg)
  File "/home/djvg/Git/X/venv/lib/python3.8/site-packages/pydantic_decoder/__init__.py", line 78, in decode
    return DecoderModel.decode(msg)
  File "/home/daan/Git/X/venv/lib/python3.8/site-packages/pydantic_decoder/__init__.py", line 120, in decode
    typ = get_model(d.pop(cls.DECODER_KEY_NAME))
KeyError: 'DECODER_MODEL_TYPE'
```

**Comment by justengel** [comment]
Ah, I see the issue. The intent of this library is to inherit from the `DecoderModel` instead of the `pydantic.BaseModel`. The `DecoderModel` has `DECODER_KEY_NAME` attribute and overrides the `dict` method to put this value in the dictionary. Without these two things the library will not work, so you do need to inherit from the `DecoderModel`.

```python
from pydantic_decoder import DecoderModel, register_model, encode, decode


class MyModel(DecoderModel):
    first_name: str
    last_name: str


class MyModel2(DecoderModel):
    first_name: str
    last_name: str


# Subclasses of the DecoderModel are registered automatically
# register_model(MyModel)
# register_model('Model2', MyModel2)

model = MyModel(first_name='John', last_name='Doe')
msg = encode(model)
m = decode(msg)
assert m == model
assert isinstance(m, MyModel)

model2 = MyModel2(first_name='John', last_name='Doe')
msg = encode(model2)
m2 = decode(msg)
assert m2 == model2
assert isinstance(m2, MyModel2)

```

**Comment by TheDJVG** [comment]
I'm not sure if I understand. Your example states:

> Use your current models without any changes.

with `register_model` so that way I would always be based of `BaseModel`? And If I would be based of `DecoderModel` there's no need for `register_model`, correct?

**Comment by justengel** [comment]
Oh, sorry, my README.rst was incorrect. My test cases were written one way and the readme was written a different way. In the past I might have automatically overridden the `dict` method and `DECODER_KEY_NAME` attribute, but that is not a good way of doing things. The documentation is fixed in version 1.0.1

You should just inherit from the `DecoderModel`. You do not need to call `register_model`. You cannot use `register_model` by itself unless you override the `dict` method and have an attribute `DECODER_KEY_NAME`.

In the end I never used this library for anything. I just couldn't get pydantic to work the way I wanted it. I also created the `serial_json` library which gave me more control over properties.
